src/exchangeRateRouter.ts

```diff
@@ -24,56 +24,82 @@ export type ExchangeRateReq = ReturnType<typeof asExchangeRateReq>
 
 const EXCHANGE_RATES_BATCH_LIMIT = 100
 
-export const exchangeRateRouter = (
-  config: ReturnType<typeof asConfig>
-): express.Router => {
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.couchUri)
+const dbRates: nano.DocumentScope<DbDoc> = nanoDb.db.use('db_rates')
+promisify(dbRates)
+
+// Query params:
+//  currency_pair: String with the two currencies separated by an underscore. Ex: "ETH_iso:USD"
+//  date: String (optional) ex. "2019-11-21T15:28:21.123Z"
+
+const exchangeRateCleaner = (req: any, res: any, next: Function): void => {
+  const currencyPair = req.query.currency_pair
+  const date = req.query.date
+  try {
+    req.requestedRates = asExchangeRatesReq({
+      data: [{ currency_pair: currencyPair, date }]
+    })
+  } catch (e) {
+    return res.status(400).send(`Missing Request fields.`)
+  }
+  next()
+}
+
+//  Query body:
+//  { data: [{ currency_pair: string, date? string) }] }
+
+const exchangeRatesCleaner = (req: any, res: any, next: Function): void => {
+  try {
+    req.requestedRates = asExchangeRatesReq(req.body)
+  } catch (e) {
+    return res.status(400).send(`Missing Request fields.`)
+  }
+  if (req.requestedRates.data.length > EXCHANGE_RATES_BATCH_LIMIT) {
+    return res
+      .status(400)
+      .send(`Exceeded Limit of ${EXCHANGE_RATES_BATCH_LIMIT}`)
+  }
+  next()
+}
+
+const queryExchangeRates = async (
+  req: any,
+  res: any,
+  next: Function
+): Promise<void> => {
+  req.requestedRatesResult = await getExchangeRates(
+    req.requestedRates.data,
+    dbRates
+  )
```

Comment:
Hmm, if this context information is scoped to just these routes, then perhaps another approach to passing data through middleware is warranted. We can discuss alternative ideas.